packages/components/src/internal/components/search/SampleFinderSavedViewsMenu.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { FC, memo, useState, useEffect, useMemo } from 'react';
+import { Button, DropdownButton, MenuItem } from 'react-bootstrap';
+
+import { LoadingSpinner } from '../base/LoadingSpinner';
+import { SubMenuItem } from '../menus/SubMenuItem';
+
+import { ComponentsAPIWrapper, getDefaultAPIWrapper } from '../../APIWrapper';
+
+import { FinderReport } from './models';
+
+interface Props {
+    api?: ComponentsAPIWrapper;
+    loadSearch: (view: FinderReport) => any;
+    saveSearch: (saveCurrentName?: boolean) => any;
+    manageSearches: () => any;
+    currentView?: FinderReport;
+    hasUnsavedChanges?: boolean;
+    sessionViewName?: string;
+    key: any;
+}
+
+export const SampleFinderSavedViewsMenu: FC<Props> = memo(props => {
+    const { api, loadSearch, manageSearches, saveSearch, currentView, hasUnsavedChanges, sessionViewName, key } = props;
+
+    const [savedSearches, setSavedSearches] = useState<FinderReport[]>(undefined);
+
+    useEffect(() => {
+        (async () => {
+            const views = await api.samples.loadFinderSearches();
+            setSavedSearches(views);
+        })();
+    }, [key]);
+
+    const hasSavedView = savedSearches?.length > 0;
```

Comment:
useMemo?

Reply:
Since it's a boolean result and this component is already using `memo()` I don't think it's necessary to use `useMemo` here.